src/components/SyncIndicator.tsx

```diff
@@ -0,0 +1,191 @@
+import { next as A } from "@automerge/automerge";
+import { DocHandle } from "@automerge/automerge-repo";
+import { useState, useEffect } from "react";
+import { useRepo } from "@automerge/automerge-repo-react-hooks";
+import { WifiIcon, WifiOffIcon } from "lucide-react";
+
+// if we are connected to the sync server and have outstanding changes we should switch to the error mode if we haven't received a sync message in a while
+// this variable specifies this timeout duration
+const SYNC_TIMEOUT = 3000 
+
+// wait initially if we haven't connnected to the sync server yet
+const INITIAL_CONNECTION_WAIT = 2000
+
+export const SyncIndicator = ({ handle }: { handle: DocHandle<unknown> }) => {
+  // todo: sync shouldn't flicker. It should be based on if we are making progress syncing the changes.
```

Comment:
@paulsonnentag I think this todo comment is out of date and can be removed, yes?